apps/vmq_commons/src/packet.erl

```diff
@@ -103,6 +107,56 @@ do_client_connect(ConnectPacket, ConnackPacket, Opts) ->
             ConnectError
     end.
 
+receive_at_most_n_frames(Socket, N, QoS) ->
```

Comment:
Since we want to receive only publish frames and ack QoS 1 accordingly, I would recommend to rename this fn.